### ISSUE TITLE
EFS Config Sanity Failure

### ISSUE THREAD

**Issue by sean-smith** [aws/aws-parallelcluster#997]
Cluster with EFS Filesystem fails during `config_sanity` check:

```
Config sanity error on resource EFSFSId: There is an existing Mount Target fsmt-543c4b1e in the Availability Zone us-east-1d for EFS fs-89d4417d2, and it does not have a security group with inbound and outbound rules that support NFS. Please modify the Mount Target's security group, or delete the Mount Target.
```

Note, the filesystem has availible mount targets in all az's. It seems the issue is with the security group check [1]:

```python
self.__check_sg_rules_for_port(rule, 2049):
...
for ip_rule in ip_rules:
            ip = ip_rule.get("CidrIp")
            # An existing rule is valid for EFS if, it allows all traffic(0.0.0.0/0)
            # from all ports or the given port, and does not have a security group restriction
            if (not port or port == port_to_check) and ip == "0.0.0.0/0" and not group:
                is_valid = True
                break
```

If there's a specified `additional_sg`, it expects a cidr range of `0.0.0.0/0` and inbound traffic allowed in on port 2049. This is too permissive for many customers.

[1] https://github.com/aws/aws-parallelcluster/blob/develop/cli/pcluster/config_sanity.py#L67

**Comment by enrico-usai** [comment]
I think it is the same problem of https://github.com/aws/aws-parallelcluster/issues/934

**Comment by hpcpony** [comment]
The documentation (https://aws-parallelcluster.readthedocs.io/en/latest/configuration.html#efs-fs-id), and source, seems to sum things up, either make the EFS accessible to everyone in the AZ or disable sanity checks.  It's not really a bug, just that the current (2.3.1) sanity check is pretty lame.

I've been looking into hacking something locally to see if I could check the master and/or compute subnets (and port 2049) against the EFS's SG but the more I look at it, doing a really good, comprehensive, sanity check appears to get pretty involved.

Part of me thinks that for _persistent_ EFS mounts the pcluster efs section isn't really all that useful (at least for me) and doing it via post-install scripts is the preferred way to go?

**Comment by sean-smith** [comment]
@hpcpony Agreed that the current validation in our sanity check is pretty lame. Probably removing the sg validation altogether would be an improvement.

I've thought about doing more in depth validation as well, checking the sg for the correct subnet cidr, which we retrieve from the fs's subnet. This gets complicated and error prone quickly.

Let's look at the downside of not doing this check, the user will create a stack, if they don't have the correct sg setup, it'll fail in the cfn-init step, causing the stack to go into CREATE_FAILED. Then the user will need to ssh in to retrieve the error in `/var/log/cfn-init.log`, see that it failed efs mount and change their SG. This is painful and we want to avoid this.

We see efs as a core piece of infrastructure and should be easy to attach from the cluster config file. There's no real advantage to attaching via the post-install script and for most users this will lead to more debugging pain.

So in summary we need some way to validate SG that isn't solely based on `0.0.0.0/0` CIDR. I'm thinking we just validate that they have the port open and we'll assume that if they have that, they'll probably also have the correct CIDR.

**Comment by sean-smith** [comment]
I've addressed this in https://github.com/aws/aws-parallelcluster/pull/1009

New sanity check doesn't check for an open `0.0.0.0/0` but instead checks to see the port that lustre/efs uses is open on that security group. The honus is on the user to configure the source correctly.
